### PR TITLE
Refactor ScrollItemWidget to use stateful image names

### PR DESCRIPTION
**Pull request by PunkPun**
Split from #20200

I cache the image collection as well. I figure that the added method in WidgetUtils will be great for other UI elements as well

### CODE REVIEW

**Comment by PunkPun** [comment]
This refactor is needed to fix a hack in MissionBrowser. To make separators stand out more they all were selected, but that meant that I couldn't use ScrollToSelectedItem in #20200. With this PR now they properly use their own sprite and don't need to be preselected

File: OpenRA.Mods.Common/Widgets/WidgetUtils.cs [hunk 1]
```diff
@@ -41,6 +41,18 @@ public static CachedTransform<(bool Disabled, bool Pressed, bool Hover, bool Foc
 					});
 		}
 
+		// TODO: refactor buttons and related UI to use this function
+		public static CachedTransform<(bool Disabled, bool Pressed, bool Hover, bool Focused, bool Highlighted), Sprite[]> GetCachedStatefulCollectionImage(string collection)
```
Comment (dragunoff) [inline OpenRA.Mods.Common/Widgets/WidgetUtils.cs]: This deals specifically with panel images so I suggest naming it something like `GetCachedStatefulPanelImages`.

File: OpenRA.Mods.Common/Widgets/ScrollItemWidget.cs [hunk 2]
```diff
@@ -20,12 +21,15 @@ public class ScrollItemWidget : ButtonWidget
 		public readonly bool EnableChildMouseOver = false;
 		public string ItemKey;
 
+		readonly CachedTransform<(bool, bool, bool, bool, bool), Sprite[]> getCollectionCache;
```
Comment (dragunoff) [inline OpenRA.Mods.Common/Widgets/ScrollItemWidget.cs]: Rename to something like `getPanelImageCollection` or `getPanelImages` to make it clear what collection it holds?

**Review by dragunoff — CHANGES_REQUESTED** [review]
LGTM - just a few naming nits. Perhaps `BaseName` is also not a great name for the image collection but I'm not sure whether we should change that now and what that name could be.

**Comment by PunkPun** [comment]
Yeah, `BaseName` isn't the best, I wanted to change that but it turned out it's quite a lot of work. Something for a future refactor. I wanted to keep this PR smaller so it could be merged quickly

File: OpenRA.Mods.Common/Widgets/WidgetUtils.cs (hunk 1 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Widgets/WidgetUtils.cs]: I don't even know what `Panel` is referring to and why that name is even used. It's why I've written `Collection` everywhere

Comment (dragunoff) [reply OpenRA.Mods.Common/Widgets/WidgetUtils.cs]: A panel is basically a rectangle built out from a collection of images - the 4 corners, 4 sides and the fill. It's how backgrounds for widgets are built.

Comment (PunkPun) [reply OpenRA.Mods.Common/Widgets/WidgetUtils.cs]: thx

File: OpenRA.Mods.Common/Widgets/WidgetUtils.cs [hunk 3]
```diff
@@ -41,6 +41,18 @@ public static CachedTransform<(bool Disabled, bool Pressed, bool Hover, bool Foc
 					});
 		}
 
+		// TODO: refactor buttons and related UI to use this function
+		public static CachedTransform<(bool Disabled, bool Pressed, bool Hover, bool Focused, bool Highlighted), Sprite[]> GetCachedStatefulPanelImage(string collection)
```
Comment (dragunoff) [inline OpenRA.Mods.Common/Widgets/WidgetUtils.cs]: Why not `GetCachedStatefulPanelImages` (plural)?

**Review by dragunoff — APPROVED** [review]

File: mods/cnc/chrome/missionbrowser.yaml [hunk 4]
```diff
@@ -24,6 +24,7 @@ Container@MISSIONBROWSER_PANEL:
 					Height: PARENT_BOTTOM - 30
 					Children:
 						ScrollItem@HEADER:
+							BaseName: scrollheader
```
Comment (RoosterDragon) [inline mods/cnc/chrome/missionbrowser.yaml:27]: Grepping for `ScrollItem@HEADER` reveals most have a `BaseName` set, but a handful do not. Do they also need updating?

- `mods\cnc\chrome\dialogs.yaml` (2 instances)
- `mods\cnc\chrome\multiplayer-browserpanels.yaml`
- `mods\cnc\chrome\replaybrowser.yaml`

**Review by RoosterDragon — COMMENTED** [review]
LGTM modulo question about possibly missing BaseNames.

File: mods/cnc/chrome/missionbrowser.yaml (hunk 4 above)
Comment (PunkPun) [reply mods/cnc/chrome/missionbrowser.yaml:27]: They do not. The other mods were already  using `scrollheader` for separators, TD was an exception as the background image matched the regular scroll item artwork. But with this refactor it now needs the custom BaseName as well

**Review by RoosterDragon — APPROVED** [review]

**Review by penev92 — APPROVED** [review]
Both @RoosterDragon and @dragunoff  approved.

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/9f873081af61f653510332801a47c36298290599...d68e69565b41a9b59619c982967a338701a3609a)

**Pull request merged**